lib/utility/logger.rb

```diff
@@ -23,7 +24,7 @@ def level=(log_level)
       end
 
       def logger
-        @logger ||= Settings[:ecs_logging] ? EcsLogging::Logger.new(STDOUT) : ::Logger.new(STDOUT)
+        @logger ||= defined?(::Settings) && ::Settings[:ecs_logging] ? EcsLogging::Logger.new(STDOUT) : ::Logger.new(STDOUT)
```

Comment:
IDK how this wasn't causing tests to fail all over the place. `Settings` doesn't seem to be defined in our test scope. Is that expected, @artem-shelkovnikov @wangch079? Maryna and I were stumped.

Reply:
I'm not quite sure - I've never had a failing test that failed due to this line.

I feel that it's better to keep the code as it is right now and see the tests - maybe mock more stuff in tests if they fail

Reply:
@artem-shelkovnikov note that it wasn't `::Settings` - that's a change from @seanstory - it was `Settings` before. Which makes me think it was some kind of a global object/variable introduced (maybe?) by our app configuration gem, but we couldn't find anything like this, we only found the [Config global variable](https://github.com/elastic/connectors-ruby/blob/main/lib/app/config.rb#L124). So we weren't sure what that top-level `Settings` object is, and what initializes it.

Reply:
Yes, it is a global object introduced by Ruby Config ([ref](https://github.com/rubyconfig/config#accessing-the-settings-object)):

>After installing the gem, `Settings` object will become available globally